identity/src/main/java/com/android/identity/L2CAPServer.java

```diff
@@ -201,14 +201,28 @@ private void closeServerSocket() {
             // Ignoring this error
             Log.e(TAG, " Error closing server socket connection " + e.getMessage(), e);
         }
-        try {
-            if (isConnected()) {
-                mSocket.close();
-                mSocket = null;
-            }
-        } catch (IOException e) {
-            // Ignoring this error
-            Log.e(TAG, " Error closing server socket connection " + e.getMessage(), e);
+        if (isConnected()) {
+            // Defer closing the socket for 1 second. This is to work around a bug where
+            // data being written and flushed - as we do in writeToSocket() above - ends
+            // up not being sent.
+            final BluetoothSocket socket = mSocket;
+            Thread closingThread = new Thread() {
+                @Override
+                public void run() {
+                    try {
+                        Thread.sleep(1000);
```

Comment:
Since we're in evil hack land, may as well bump this to 10s or more.

It's in the background, so it's not blocking anything to give us extra buffer.